### PR TITLE
Intel MPI U6

### PR DESCRIPTION
**Pull request by sean-smith**
Upgrade to Intel MPI U6.

Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: recipes/intel_mpi.rb [hunk 1]
```diff
@@ -39,7 +39,10 @@
   cwd node['cfncluster']['sources_dir']
   code <<-INTELMPI
     set -e
-    ./aws_impi.sh install -check_efa 0 -version #{node['cfncluster']['intelmpi']['version']}
+    tar -xf l_mpi_#{node['cfncluster']['intelmpi']['version']}.tgz
```
Comment (demartinofra) [inline recipes/intel_mpi.rb]: can we fetch this as part of a remote_file directive and have it downloaded in the sources dir or tmp dir?
you could just update the part above that downloads the installer which is not used anymore.

Comment (sean-smith) [reply recipes/intel_mpi.rb]: This does download into the sources directory already. Not sure what you mean.

**Comment by sean-smith** [comment]
Yes, I handled that in https://github.com/aws/aws-parallelcluster-cookbook/pull/487

File: recipes/intel_mpi.rb (hunk 1 above)
Comment (demartinofra) [reply recipes/intel_mpi.rb]: I think you can ignore my comment :)

File: recipes/intel_mpi.rb [hunk 2]
```diff
@@ -39,10 +33,19 @@
...
+    cd l_mpi_#{node['cfncluster']['intelmpi']['version']}/
+    ./install.sh -s silent.cfg --accept_eula
   INTELMPI
-  creates '/opt/intel/impi'
+  creates "/opt/intel/impi/#{node['cfncluster']['intelmpi']['version']}"
+end
+
+bash "create modulefile" do
+  cwd node['cfncluster']['modulefile_dir']
+  code <<-MODULEFILE
```
Comment (demartinofra) [inline recipes/intel_mpi.rb]: set -e flag or use command

Comment (sean-smith) [reply recipes/intel_mpi.rb]: done

File: recipes/tests.rb [hunk 3]
```diff
@@ -136,11 +136,27 @@
       command "rpm -qa | grep libfabric && rpm -qa | grep efa-"
       user node['cfncluster']['cfn_cluster_user']
     end
+    execute 'check intel mpi installed' do
```
Comment (demartinofra) [inline recipes/tests.rb]: Leaving the comment here but it refers to L132: Aren't we installing EFA and intel mpi in cn regions too?

File: recipes/tests.rb [hunk 4]
```diff
@@ -136,11 +136,27 @@
...
   when 'ubuntu1604', 'ubuntu1804'
     execute 'check efa rpm installed' do
       command "dpkg -l | grep libfabric && dpkg -l | grep 'efa '"
       user node['cfncluster']['cfn_cluster_user']
     end
+    execute 'check intel mpi installed' do
+      command "dpkg -l | grep intel-mpi"
+      user node['cfncluster']['cfn_cluster_user']
+    end
+    execute 'check intel mpi version' do
```
Comment (demartinofra) [inline recipes/tests.rb]: minor: you could move this outside of when and avoid the duplication

File: recipes/tests.rb (hunk 3 above)
Comment (sean-smith) [reply recipes/tests.rb]: yes, I've removed not `cn` check

File: recipes/tests.rb (hunk 4 above)
Comment (sean-smith) [reply recipes/tests.rb]: done

**Pull request merged**